src/foundry/client/pixi/placeables/primary-canvas-objects/primary-canvas-object.d.mts

```diff
@@ -7,148 +7,181 @@ declare class PrimaryCanvasObject {
+   * The update ID of canvas transform matrix.
+   */
+  protected _canvasTransformID: number;
 
-    /** The width of the PCO */
-    width: number;
+  /**
+   * The canvas bounds of this object.
+   */
+  canvasBounds: PIXI.Rectangle;
 
-    /** The height of the PCO */
-    height: number;
+  /**
+   * The canvas bounds of this object.
+   */
+  protected _canvasBounds: PIXI.Bounds;
 
-    /** The alpha of this PCO */
-    alpha: number;
+  /**
+   * The update ID of the canvas bounds.
+   * Increment to force recalculation.
+   */
+  protected _canvasBoundsID: number;
 
-    /** The rotation of this PCO */
-    rotation: number;
+  /**
+   * Calculate the canvas bounds of this object.
+   */
+  protected _calculateCanvasBounds(): void;
 
-    /** The PCO is hidden? */
-    hidden: boolean;
+  /**
+   * Recalculate the canvas transform and bounds of this object and its children, if necessary.
+   */
+  updateCanvasTransform(): void;
 
-    /** The elevation of the PCO */
-    elevation: number;
+  /**
+   * Called when the canvas transform changed.
+   */
+  protected _onCanvasTransformUpdate(): void;
 
-    /** The sort key that resolves ties among the same elevation */
-    sort: number;
+  /**
+   * Called when the canvas bounds changed.
+   */
+  protected _onCanvasBoundsUpdate(): void;
 
-    /** The data texture values */
-    texture: PIXI.RenderTexture;
-  }
+  /**
+   * Is the given point in canvas space contained in this object?
+   * @param point - The point in canvas space.
+   */
+  containsCanvasPoint(point: PIXI.IPointData): boolean;
+}
 
+declare global {
   /**
    * A mixin which decorates a DisplayObject with additional properties expected for rendering in the PrimaryCanvasGroup.
    * @param DisplayObject - The parent DisplayObject class being mixed
    * @returns A DisplayObject subclass mixed with PrimaryCanvasObject features
    */
   function PrimaryCanvasObjectMixin<BaseClass extends PrimaryCanvasObjectMixin.BaseClass>(
     DisplayObject: BaseClass,
-  ): Mixin<typeof PrimaryCanvasObject, BaseClass>;
+  ): Mixin<typeof PrimaryCanvasObject, ReturnType<typeof CanvasTransformMixin<BaseClass>>>;
 
   namespace PrimaryCanvasObjectMixin {
     type BaseClass = PrimaryCanvasObjectMixinBaseClass;
 
     type MixinClass = typeof PrimaryCanvasObject;
   }
+
+  /**
+   * A mixin which decorates a DisplayObject with additional properties for canvas transforms and bounds.
+   * @param DisplayObject - The parent DisplayObject class being mixed
+   * @returns A DisplayObject subclass mixed with CanvasTransformMixin features
+   */
+  function CanvasTransformMixin<BaseClass extends PrimaryCanvasObjectMixinBaseClass>(
+    DisplayObject: BaseClass,
+  ): Mixin<typeof CanvasTransformMixinClass, BaseClass>;
+
+  namespace CanvasTransformMixin {
+    type MixinClass = typeof CanvasTransformMixinClass;
+  }
 }
 
 type DisplayObjectClass = typeof DisplayObject;
 
+// TODO: Is this advanced construction still needed with the rework to PCO?
```

Comment:
This construction is basically an alternative to `DisplayObjectClass.AnyConstructor`. It was designed before I settled on the format of `declare abstract class AnyDisplayObjectClass extends DisplayObjectClass { constructor(arg0: never, ...args: never[]) }`

You can move it over to the standard `AnyConstructor` format if you want. Keep the comment where the mixin uses it though.